### PR TITLE
[Settings - Themes]: Implementing dropdown in settings to change themes

### LINKED ISSUE #207: Switching between dark and light theme with ctrl-shift-p stopped working
Before the latest update, I was able to switch between the dark and the light theme. This does not seem to work anymore (nothing happens when switching):

image

### PR DESCRIPTION
**Pull request by mliao95**
This PR adds a settings dropdown to change the themes of the DevTools extension.
image

This PR also broadens the use of the settingsProvider (formerly tabSettingsProvider) to include the theme settings.

Closes #207

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -20,5 +20,6 @@
         "userdatadir",
         "vsce",
         "vsix"
-    ]
+    ],
+    "webview.experimental.useIframes": true
```
Comment (vidorteg) [inline .vscode/settings.json:24]: remove this.

File: src/test/helpers.ts [hunk 2]
```diff
@@ -62,6 +62,8 @@ export function createFakeVSCode() {
                     switch(name) {
                         case "enableNetwork":
                             return true;
+                        case "themeString":
```
Comment (vidorteg) [inline src/test/helpers.ts:65]: Lets add an interface for this values.

**Review by vidorteg — APPROVED** [review]
LGTM % comments

File: src/host/polyfills/simpleView.ts [hunk 3]
```diff
@@ -357,3 +357,50 @@ export function applyRemoveNonSupportedRevealContextMenu(content: string) {
...
+    const injectFunctionsPattern = /async _createAppUI/;
+    if (content.match(injectFunctionsPattern)) {
+        content = content.replace(injectFunctionsPattern, `${getThemes.toString().slice(9)} getThemePromise(){
+            const promise = new Promise(function(resolve){
+                this.getThemes((object)=>{
+                  const theme = object.theme;
+                  console.log('Theme =' + theme);
+                  resolve(theme);
+                });
+              }.bind(this));
```
Comment (mliao95) [inline src/host/polyfills/simpleView.ts:393]: @vidorteg - I checked, and we do need this `bind` call to access the `getThemes` function

**Comment by mliao95** [comment]
In the most recent patch, I did some things to clean up the code:
- Differentiated between `themeString` and `theme` in both the extension code and injected code
    - `themeString` will be the user-visible string (e.g. "System preference") and the `theme` will contain the id string (e.g. "systemPreferred")
- Moved the switch logic into `toolsHost.ts`
    - seems to make more sense to send over the needed `theme` string rather than having logic to parse it injected into the source code.

**Review by vidorteg — APPROVED** [review]

**Pull request merged**